lms/djangoapps/mobile_api/courses/views.py

```diff
@@ -0,0 +1,126 @@
+                "id": "org.4/course_4/Run_4",
+                "latest_updates": {
+                    "video": "None"
+                },
+                "end": "None",
+                "name": "Run 4",
+                "course_handouts": "http://testserver/api/mobile/v0.5/course_info/org.4/course_4/Run_4/handouts",
+                "start": "2030-01-01 00:00:00",
+                "course_image": "/c4x/org.4/course_4/asset/images_course_image.jpg"
+                }
+            },
+            ...
+        ]
+    """
+
+    authentication_classes = (OAuth2Authentication, SessionAuthentication)
+    permission_classes = (permissions.IsAuthenticated,)
+    serializer_class = serializers.CoursesWithFriendsSerializer
+
+
+    def list(self, request, *args, **kwargs):
+        serializer = self.get_serializer(data=request.GET, files=request.FILES)
+        if serializer.is_valid():
+            # Get friends from Facebook
+            graph = facebook.GraphAPI(serializer.object['oauth_token'])
+            friends = graph.request(_FACEBOOK_API_VERSION + "me/friends")
+            data = self.get_pagination(friends)
+            # Get the linked edX user if it exists
+            friends_that_are_edX_users = []
+            for friend in data:
+                query_set = UserSocialAuth.objects.filter(uid=unicode(friend['id']))
+                if query_set.count() == 1: 
+                    friend['edX_id'] = query_set[0].user_id
+                    friend['edX_username'] = query_set[0].user.username
+                    friends_that_are_edX_users.append(friend)
+            # Filter by sharing preferences
+            friends_that_are_edX_users_with_sharing = [friend for friend in friends_that_are_edX_users if self.sharing_pref_true(friend)]
+            # Get unique courses
+            enrollments = self.get_unique_courses(friends_that_are_edX_users_with_sharing)
+            # Get course objects 
+            courses = [enrollment for enrollment in enrollments if enrollment.course and is_mobile_available_for_user(self.request.user, enrollment.course)]
+            return Response(CourseEnrollmentSerializer(courses, context={'request' : request}).data)
+        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+
+    def get_unique_courses(self, friends_that_are_edX_users_with_sharing):
+        '''
+            Return a list of unique courses friends_that_are_edX_users_with_sharing are in
+        '''
+        enrollments = []
+        for friend in friends_that_are_edX_users_with_sharing:
+            query_set = CourseEnrollment.objects.filter(user_id = friend['edX_id'], is_active = True)
+            if query_set.count() > 0:
+                for i in range(len(query_set)):
+                    if not self.is_member(enrollments, query_set[i]):
+                        enrollments.append(query_set[i])
+        return enrollments
+
+    def get_pagination(self, friends): 
```

Comment:
That is, what is the default page size for the facebook Graph API?  Is the page size a parameter that can be tweaked? 
If a user has a huge number of friends, we will be calling the FB API num_friends/page_size times.  
However, on the app, we don't plan to display all those friends anyway, right?
If we do, for scalability, this API itself would need to be paginated.
We'll need to carefully consider this - and think about a strategy such as caching/pagination/etc.